pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -584,6 +684,28 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         }
     }
 
+    function _exitRemoveToken(address sender, bytes memory userData)
+        private
+        view
+        whenNotPaused
+        returns (uint256 bptAmountIn, uint256[] memory amountsOut)
+    {
+        // This exit function is disabled if the contract is paused.
+
+        // This exit function can only be called by the Pool itself - the authorization logic that governs when that
+        // call can be made resides in withdrawCollectedManagementFees.
+        _require(sender == address(this), Errors.UNAUTHORIZED_EXIT);
+
+        // No BPT is required for the exit operation itself. The `removeToken` function calculates and returns
+        // the bptAmountIn, but leaves any further action, such as burning BPT, up to the caller.
+        bptAmountIn = 0;
+
+        (uint256 tokenIndex, uint256 amountOut) = userData.removeToken();
+
+        amountsOut = new uint256[](_getTotalTokens());
```

Comment:
Here is were we need to upscale, as we do in all other join/exit handlers. That upscale will cancel out the downscale that BasePool does when we return (but more generally, inside these handlers we always work with upscaled amounts).